seed/views/column_mapping_presets.py

```diff
@@ -67,24 +70,50 @@ def update(self, request, pk=None):
               paramType: body
         """
         try:
-            updated = ColumnMappingPreset.objects.filter(pk=pk).update(**request.data)
-        except FieldDoesNotExist as e:
+            preset = ColumnMappingPreset.objects.get(pk=pk)
+        except ColumnMappingPreset.DoesNotExist:
             return JsonResponse({
                 'status': 'error',
-                'message': str(e),
+                'data': 'No preset with given id'
             }, status=HTTP_400_BAD_REQUEST)
 
-        if updated:
-            preset = ColumnMappingPreset.objects.get(pk=pk)
-            status = 'success'
-            data = ColumnMappingPresetSerializer(preset).data
-        else:
-            status = 'error'
-            data = "Column Mapping Preset not updated."
+        if preset.preset_type == ColumnMappingPreset.BUILDINGSYNC_DEFAULT:
+            return JsonResponse({
+                'status': 'error',
+                'data': 'Default BuildingSync presets are not editable'
+            }, status=HTTP_400_BAD_REQUEST)
 
+        updated_name, updated_mappings = request.data.get('name'), request.data.get('mappings')
+
+        # update the name
+        if updated_name is not None:
+            preset.name = updated_name
+
+        # update the mappings according to the preset type
+        if updated_mappings is not None:
+            if preset.preset_type == ColumnMappingPreset.BUILDINGSYNC_CUSTOM:
+                # only allow these updates to the mappings
+                # - changing the to_field
```

Comment:
Since we only allow creating Properties from Bsync files, I'd say no to `to_table_name`.
I'm not sure if we want to allow users to change the `from_units`. Thoughts @nllong ?

Reply:
Makes sense. If we do end up not allowing these changes, I'd vote to disable changing this options in the FE for any BS preset (default or custom).

Reply:
Decided that users must be able to change the from_units for custom presets (still not allowed for default bsync presets). This is necessary because users might want to change a mapping to a SEED column that uses different units (maybe strange case but possible).

Reply:
Added this in 917384e

Test by creating a custom preset, changing the units, saving it, then REFRESHING the page and viewing the preset again (force it to reload the presets from the backend). The units should have been updated.